ui/src/app/components/sharded-dataflow-migration-details-form/sharded-dataflow-migration-details-form.component.ts

```diff
@@ -0,0 +1,396 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { FormArray, FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { MigrationDetails, Profile, StorageKeys } from 'src/app/app.constants';
+import IDbConfig from 'src/app/model/db-config';
+import IConnectionProfile, { ICreateConnectionProfile, ICreateConnectionProfileV2, IDataShard, IDatastreamConnProfile, IDirectConnectionConfig, ILogicalShard, IMigrationProfile, ISetUpConnectionProfile, IShardConfigurationDataflow, IShardedDataflowMigration } from 'src/app/model/profile';
+import { FetchService } from 'src/app/services/fetch/fetch.service';
+import { SnackbarService } from 'src/app/services/snackbar/snackbar.service';
+
+@Component({
+  selector: 'app-sharded-dataflow-migration-details-form',
+  templateUrl: './sharded-dataflow-migration-details-form.component.html',
+  styleUrls: ['./sharded-dataflow-migration-details-form.component.scss']
+})
+export class ShardedDataflowMigrationDetailsFormComponent implements OnInit {
+
+  migrationProfileForm!: FormGroup
+  selectedProfile: string = ''
+  profileType: string = ''
+  sourceProfileList: IConnectionProfile[] = []
+  targetProfileList: IConnectionProfile[] = []
```

Comment:
Why do we need src and target connection profiles here ? Will only target not suffice ?

Reply:
Didn't get you, we need both source and target connection profile arrays to configure each shard, right?